src/mlpack/core/data/load_csv.cpp

```diff
@@ -25,15 +25,35 @@ LoadCSV::LoadCSV(const std::string& file) :
   CheckOpen();
 
   // Set rules.
-  if (extension == "csv" || extension == "txt")
+  if (extension == "csv")
   {
+    // Match quoted strings as: "string" or 'string'
     // Match all characters that are not ',', '\r', or '\n'.
-    stringRule = qi::raw[*~qi::char_(" ,\r\n")];
+    stringRule = qi::raw[(qi::char_("'") >> *((qi::char_ - "'") |
+                                 "'" >> qi::char_("'")) >> "'") |
+                         (qi::char_('"') >> *((qi::char_ - '"') |
+                                 '"' >> qi::char_('"')) >> '"') |
+                         *~qi::char_(",\r\n")];
+  }
+  else if (extension == "txt")
```

Comment:
Yeah they are almost same, except for txt has space delimiter. I made it different so that it would be easier to add something to `csv` later,  without changing `txt`.

Reply:
So perhaps just to avoid a bunch of duplicated code, we could just set the delimiter in each case and use the same routine afterwards? let me know what you think.

Reply:
yeah I'll do that